lib/scope_file_access.rb

```diff
@@ -79,17 +79,23 @@ def tarball_path(system_file)
     end
   end
 
-  def file_content(filename)
+  def file_content(system_file)
     if !extracted
-      raise Machinery::Errors::FileUtilsError, "The requested file '#{filename}' is not available" \
-        " because files for scope '#{scope_name}' were not extracted."
+      raise Machinery::Errors::FileUtilsError, "The requested file '#{system_file.name}' is not" \
+        " available because files for scope '#{scope_name}' were not extracted."
     end
 
     tarball_path = File.join(scope_file_store.path, "files.tgz")
     begin
-      Cheetah.run("tar", "xfO", tarball_path, filename.gsub(/^\//, ""), stdout: :capture)
+      Cheetah.run("tar", "xfO", tarball_path, system_file.name.gsub(/^\//, ""), stdout: :capture)
     rescue
-      raise Machinery::Errors::FileUtilsError, "The requested file '#{filename}' was not found."
+      raise Machinery::Errors::FileUtilsError,
+        "The requested file '#{system_file.name}' was not found."
     end
   end
+
+  def binary?(system_file)
+    output = Cheetah.run("file", "-", stdin: file_content(system_file), stdout: :capture)
+    !output.include?("ASCII") && !output.include?("no read permission")
```

Comment:
Machinery should always be able to access its filestore, no? I think I'd rather treat the `no read permission` case as an error.